src/pages/AdminPages/UnapprovedAccounts.js

```diff
@@ -0,0 +1,115 @@
+import React, {useEffect, useState} from 'react'
+import { useSelector, useDispatch, shallowEqual } from 'react-redux'
+import Authenticate from '../../components/Authenticate'
+import PageNotFound from '../PageNotFound'
+import { getUnapprovedUsers } from "../../store/reducers/unapprovedUser";
+import { Table } from '../../components/Table/Table';
+import BaseModal from '../../components/Modal/BaseModal';
+
+const UnapprovedAccounts = () => {
+    const dispatch = useDispatch()
+    const [selectedUser, setSelectedUser] = useState({})
+    const [selectedRole, setSelectedRole] = useState({
+        isAdmin : false,
+        isSuperAdmin : false,
+        isUser : true,
+        isAffiliate : false,
+        isTrader : false
+    })
+    const [showModal, setShowModal] = useState(false)
+    const { role, isLoading, unapprovedUsers } = useSelector(state => ({
+        role : state.user.role,
+        isLoading : state.isLoading,
+        unapprovedUsers: state.unapprovedUsers
+    }), shallowEqual)
+
+    const setRole = e => {
+        console.log(e.target.value)
+        e.preventDefault();
+        const data = {
+            isAdmin : false,
+            isSuperAdmin : false,
+            isUser : false,
+            isAffiliate : false,
+            isTrader : false
+        }
+        if (e.target.value === 'admin') {
+            data.isAdmin = true;
+        } 
+        if (e.target.value === 'superadmin') {
+            data.isSuperAdmin = true;
+        }
+        if (e.target.value === 'user') {
+            data.isUser = true
+        }
+        if (e.target.value === 'affiliate') {
+            data.isAffiliate = true
+        }
+        if (e.target.value === 'trader') {
+            data.isTrader = true
+        }
```

Comment:
1. You are presuming that the admin will not first click / accidentally choose something else before choosing the right role...You need to set the other roles to false when he picks  any role.

2. superadmin is also an admin so both will be true when superadmin is chosen. an affiliate is also a user so both will be true when affiliate is chosen

Reply:
Alright, thanks for this.